clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsServerClusterProvider.java

```diff
@@ -0,0 +1,269 @@
+
+        if (clusterMap.get(metadata.applicationName) != null) {
+          clusterMap.get(metadata.applicationName).add(spinnakerCluster);
+        } else {
+          clusterMap.put(metadata.applicationName, Sets.newHashSet(spinnakerCluster));
+        }
+      }
+    }
+
+    return clusterMap;
+  }
+
+  private Set<com.netflix.spinnaker.clouddriver.model.LoadBalancer> extractLoadBalancersData(DescribeLoadBalancersResult loadBalancersResult) {
+    Set<com.netflix.spinnaker.clouddriver.model.LoadBalancer> loadBalancers = Sets.newHashSet();
+    for (LoadBalancer elb: loadBalancersResult.getLoadBalancers()) {
+      AmazonLoadBalancer loadBalancer = new AmazonLoadBalancer();
+      loadBalancer.setName(elb.getLoadBalancerName());
+      loadBalancers.add(loadBalancer);
+    }
+    return loadBalancers;
+  }
+
+  private EcsServerCluster generateSpinnakerServerCluster(AmazonCredentials credentials, ServiceMetadata metadata, Set<com.netflix.spinnaker.clouddriver.model.LoadBalancer> loadBalancers, EcsServerGroup ecsServerGroup) {
+    return new EcsServerCluster()
+          .withAccountName(credentials.getName())
+          .withName(metadata.cloudStack)
+          .withLoadBalancers(loadBalancers)
+          .withServerGroups(Sets.newHashSet(ecsServerGroup));
+  }
+
+  private EcsServerGroup generateServerGroup(AmazonCredentials.AWSRegion awsRegion, ServiceMetadata metadata, Set<Instance> instances) {
+    return new EcsServerGroup()
+          .withName(constructServerGroupName(metadata))
+          .withCloudProvider("aws")   // TODO - Implement ECS in Deck so we can stop tricking the front-end app here
+          .withType("aws")            // TODO - Implement ECS in Deck so we can stop tricking the front-end app here
+          .withRegion(awsRegion.getName())
+          .withInstances(instances);
+  }
+
+  private String constructServerGroupName(ServiceMetadata metadata) {
+    return metadata.applicationName + "-" +  metadata.cloudStack + "-" + metadata.serverGroupVersion; // TODO - support CLOUD_DETAIL variable
+  }
+
+  private ServiceMetadata extractMetadataFromServiceArn(String arn) {
+    if (!arn.contains("/")) {
+      return null; // TODO - do a better verification,
+    }
+
+    String[] splitArn = arn.split("/");
+    if (splitArn.length != 2) {
+      return null; // TODO - do a better verification,
+    }
+
+    String[] splitResourceName = splitArn[1].split("-");
+
+    if (splitResourceName.length != 3) {
+      return null; // TODO - do a better verification, and handle cases with both cloudStack and CloudDetail
+    }
+
+    ServiceMetadata serviceMetadata = new ServiceMetadata();
+    serviceMetadata.applicationName = splitResourceName[0];
+    serviceMetadata.cloudStack = splitResourceName[1];
+    serviceMetadata.serverGroupVersion = splitResourceName[2];
+
+    return serviceMetadata;
+  }
+
+  class ServiceMetadata {
```

Comment:
Do internal classes not follow the setters/getters structure of regular classes?

Reply:
Yeah but I didn't want to litter the code here.  That being said, we should use the Lombok data annotation going forward, which auto-generates all setters and getters with the use of a simple annotation, which is a much better solution than what was done here

Reply:
Nice! that Lombok annotation removes a lot of boilerplate code